src/Api.Rest/HttpClient/RawData/RawDataServiceRestClient.cs

```diff
@@ -105,7 +105,33 @@ private async Task<RawDataInformationDto[]> ListRawDataByUuidList( RawDataEntity
 			return result.SelectMany( r => r ).ToArray();
 		}
 
-		private Task UploadRawData( RawDataInformationDto info, byte[] data, HttpMethod method, CancellationToken cancellationToken )
+		private async Task<RawDataInformationDto> CreateRawDataInternal( RawDataInformationDto info, byte[] data, HttpMethod method, CancellationToken cancellationToken )
+		{
+			StringUuidTools.CheckUuid( info.Target.Entity, info.Target.Uuid );
+
+			if( string.IsNullOrEmpty( info.FileName ) )
+				throw new ArgumentException( "FileName needs to be set.", nameof( info ) );
+
+			var requestString = info.Key is >= 0
+				? $"rawData/{info.Target.Entity}/{info.Target.Uuid}/{info.Key}"
+				: $"rawData/{info.Target.Entity}/{info.Target.Uuid}";
+
+			var stream = new MemoryStream( data, 0, data.Length, false, true );
+
+			var featureMatrix = await GetFeatureMatrixInternal( FetchBehavior.FetchIfNotCached, cancellationToken ).ConfigureAwait( false );
+			var request = RequestBuilder.CreateWithAttachment( method, requestString, stream, info.MimeType, info.Size, info.MD5, info.FileName );
+
+			if ( featureMatrix.SupportsCreateRawDataResult )
+				return await _RestClient.Request<RawDataInformationDto>( request , cancellationToken );
+
+			await _RestClient.Request( request, cancellationToken );
+
+			var entity = info.Target.Entity;
+			var rawDataList = await ListRawData( entity, new[] { info.Target.Uuid }, null, cancellationToken );
+			return rawDataList.OrderByDescending( rawData => rawData.Key ?? 0 ).First();

```

Comment:
Maybe consider FirstOrDefault for the extreme edge case where the RawData has already been deleted between creation and refetch.

Reply:
Indeed, that's an edge case which should not happen in practice. In such a case I would not return null since the identification of the problem would then be handed over to the client. The information that the list doesn't contain any key would be lost.
Another edge case (and still very unlikely) is that the raw data has already been deleted and there is another key available. Then a wrong key would be returned.
I would not handle those cases since I cannot imagine how a raw data can be deleted immediately after creation.